rockusb/examples/rockusb.rs

```diff
@@ -70,6 +76,48 @@ fn read_capability(mut transport: Transport) -> Result<()> {
     Ok(())
 }
 
+static MAX_BLOCKS_ERASE: u32 = 16;
+static MAX_DIRECT_ERASE: u32 = 1024;
+static MAX_LBA_ERASE: u32 = 32 * 1024;
+
+fn erase_flash(mut transport: Transport) -> Result<()> {
+    // Get flash info
+    let flash_info = transport.flash_info()?;
+    print_flash_info(&flash_info);
+
+    ensure!(flash_info.sectors() > 0, "Invalid flash chip");
+
+    // Get flash id
+    let flash_id = transport.flash_id()?;
+    let is_direct = flash_id.to_str() == "NAND " || flash_id.to_str() == "NOR  ";
+
+    // Get flash capability
+    let capability = transport.capability()?;
+
+    let is_lba = capability.direct_lba() || !is_direct;
+
+    let mut blocks_left = flash_info.sectors();
+    let mut first = 0;
+    let max_blocks = if is_direct {
+        MAX_DIRECT_ERASE
+    } else if is_lba {
+        MAX_LBA_ERASE
+    } else {
+        MAX_BLOCKS_ERASE
```

Comment:
This one is never used as due to the logic either `is_direct` or `is_lba` will be true; Also some comments of why this is done would be good ;)

Reply:
Indeed... The number of blocks in `MAX_DIRECT_ERASE` and others are smaller values than rkdeveloptool. The idea was to be able to show some useful progress information (as opposed to rkdeveloptool that will just show  "Done" at the end of SPI erasure). But it is not that important and could be revisited later if showing progress is required.